### ISSUE TITLE
control n’est pas reconnu par kitty sous macos

### ISSUE THREAD

**Issue by brabalan** [Nuclear-Squid/ergol#237]
Dans la discussion de ce bug https://github.com/kovidgoyal/kitty/issues/7841, il semblerait que ergo‑l soit en cause :

> I am saying your custom layout is SENDING TEXT for ctrl-a so kitty will treat that key press as generating text not as a control key. As it should.

Je ne suis pas sûr de comprendre ce que cela veut dire, mais visiblement il y a un problème avec le keylayout généré (tout fonctionne quand je passe en qwerty système).

**Comment by arobase-che** [comment]
Absolument. En tout cas sus Arch Linux, le passage en 2.44 a corrigé le problème.

**Comment by brabalan** [comment]
Je viens de réinstaller kitty pour tester, et cela ne fonctionne pas plus. Je n’avais pas xkeyboard-config d’installé, donc pour moi une nouvelle version ne change rien. Il faut que j’installe une nouvelle version de la dispo ?

Edit: je viens de récupérer la version 1.01 et c’est exactement la même que la 1.0 (`Generated by kalamine on 2024-05-25`), donc cela ne va pas résoudre mon problème. Est-ce que je réouvre cette issue ?

**Comment by trilowy** [comment]
Non c’est pas la dispo, c’est la version de xkeyboard-config si j’ai bien compris (ce qui interprète la dispo), mais c’est peut-être pas encore à jour dans ton OS. T’as quoi comme OS/version ?

**Comment by brabalan** [comment]
MacOS Tahoe 26.4. Et selon brew, xkeyboard-config n’est pas installé (ce qui n’est pas surprenant parce que X n’est pas installé).

**Comment by trilowy** [comment]
Ah oui my bad, c’est macOS. C’est uniquement sous Linux que c’est corrigé, je réouvre.

**Comment by thblt** [comment]
>> I am saying your custom layout is SENDING TEXT for ctrl-a so kitty will treat that key press as generating text not as a control key. As it should.

> Je ne suis pas sûr de comprendre ce que cela veut dire, mais visiblement il y a un problème avec le keylayout généré (tout fonctionne quand je passe en qwerty système).

C’est parce que kitty... ne reconnaît pas ctrl-a. Il ne connaît pas ce raccourci. Si les programmes Mac comprennent ctrl-a, ctrl-e etc, c’est parce que les dispositions standard fournissent explicitement une couche ctrl qui envoie "début de ligne" quand on presse ctrl-a. C’est ce que font toutes les dispos standard que j’ai testées, et aussi pas mal de non-standard comme bépo, mais pas ergol. Voici cette couche ctrl (dans Ukelele) 

<img width="747" height="333" alt="Image" src="https://github.com/user-attachments/assets/a90e2a12-fb3e-42f7-b19c-9d3f4c59ff87" />

Le problème serait à résoudre dans Kalamine a priori, en collant simplement la couche ctrl dans le template (il faut aussi ajouter le masque de modificateurs qui l’active).

**Comment by thblt** [comment]
(il y a sans doute d’autres subtilités puisque d’autres émulateurs passent tout à fait normalement le ctrl-a, mais ça devrait être une solution complete)
